File: library/src/main/java/com/felix/library/utils/HttpUtils.java
package com.felix.library.utils;

import com.felix.library.net.BaseServerResponse;
import com.felix.library.net.RestApiProvider;
import com.felix.library.net.RestBaseCallBack;
import com.felix.library.net.RestService;
import okhttp3.RequestBody;
import retrofit2.Call;

import java.util.Map;

/**
 * @author liuhaiyang
 */
public class HttpUtils {

    public static void getData(String url, Object request, RestBaseCallBack callBack) {
        Map<String, Object> queryMap = JsonUtils.javaBeanToMap(request);
        RestService mRestService = getService();
        Call<BaseServerResponse> mCall;
        mCall = mRestService.loadData(url, queryMap);
        mCall.enqueue(callBack);
    }

    public static void getData(String url, RestBaseCallBack callBack) {
        RestService mRestService = getService();
        Call<BaseServerResponse> mCall;
        mCall = mRestService.loadData(url);
        mCall.enqueue(callBack);
    }

    public static void postData(String url, Object request, RestBaseCallBack callBack) {
        RestService mRestService = getService();
        Call<BaseServerResponse> mCall;
        mCall = mRestService.postData(url, request);
        mCall.enqueue(callBack);
    }

    public static void postData(String url, RestBaseCallBack callBack) {
        postData(url, new Object(), callBack);
    }


    public static void upload(String url, Map<String, RequestBody> partMap, RestBaseCallBack callBack) {
        RestService mRestService = getService();
        Call<BaseServerResponse> mCall;
        mCall = mRestService.uploadFile(url, partMap);
        mCall.enqueue(callBack);
    }

    private static RestService getService() {
        return RestApiProvider.getInstance().builder().getApiService(RestService.class);
    }
}

File: app/src/main/java/com/felix/httputils/utils/HttpUtils.kt
package com.felix.httputils.utils

import com.felix.httputils.net.BaseServerResponse
import com.felix.httputils.net.RestApiProvider
import com.felix.httputils.net.RestBaseCallBack
import com.felix.httputils.net.RestService
import okhttp3.RequestBody
import retrofit2.Call

/**
 * Created by liuhaiyang on 2018/12/25.
 */
class HttpUtils {

    companion object {

        fun getData(url: String, request: Any, callBack: RestBaseCallBack<*>) {
            val queryMap = JsonUtils.javaBeanToMap(request)
            val service = RestApiProvider.getInstance().builder().getApiService(RestService::class.java)
            val mCall: Call<BaseServerResponse> = service.loadData(url, queryMap)
            mCall.enqueue(callBack)
        }

        fun getData(url: String, callBack: RestBaseCallBack<*>) {
            val service = RestApiProvider.getInstance().builder().getApiService(RestService::class.java)
            val mCall: Call<BaseServerResponse> = service.loadData(url)
            mCall.enqueue(callBack)
        }

        fun postData(url: String, request: Any, callBack: RestBaseCallBack<*>) {
            val service = RestApiProvider.getInstance().builder().getApiService(RestService::class.java)
            val mCall: Call<BaseServerResponse> = service.postData(url, request)
            mCall.enqueue(callBack)
        }

        fun postData(url: String, callBack: RestBaseCallBack<*>) {
            postData(url, Any(), callBack)
        }

        fun upload(url: String, partMap: Map<String, RequestBody>, callBack: RestBaseCallBack<*>) {
            val mRestService = RestApiProvider.getInstance().builder().getApiService(RestService::class.java)
            val mCall: Call<BaseServerResponse> = mRestService.uploadFile(url, partMap)
            mCall.enqueue(callBack)
        }
    }
}